cmd/clusterctl/pkg/client/repository/client.go

```diff
@@ -151,3 +155,25 @@ func repositoryFactory(providerConfig config.Provider, configVariablesClient con
 
 	return nil, errors.Errorf("invalid provider url. there are no provider implementation for %q schema", rURL.Scheme)
 }
+
+const overrideFolder = "overrides"
+
+// getLocalOverride return local override file from the config folder, if it exists.
+// This is required for development purposes, but it can be used also in production as a workaround for problems on the official repositories
+func getLocalOverride(provider config.Provider, version, path string) ([]byte, error) {
+
+	// local override files are searched at $home/cluster-api/overrides/<provider-name>/<version>/<path>
+	homeFolder := filepath.Join(homedir.HomeDir(), config.ConfigFolder)
+	overridePath := filepath.Join(homeFolder, overrideFolder, provider.Name(), version, path)
+
+	// it the local override exists use it, otherwise read from the repository
+	if _, err := os.Stat(overridePath); err == nil {
+		content, err := ioutil.ReadFile(overridePath)
+		if err != nil {
+			return nil, errors.Wrapf(err, "failed to read local override for %s/%s/%s", provider.Name(), version, path)
+		}
+		return content, nil
+	}
```

Comment:
Shouldn't we check if the error from `Stat`?

Reply:
done!